rom/dev/src/main.rs

```diff
@@ -154,6 +168,20 @@ fn rom_panic(_: &core::panic::PanicInfo) -> ! {
     report_error(CaliptraError::ROM_GLOBAL_PANIC.into());
 }
 
+fn report_error_update_reset(code: u32) {
+    cprintln!("ROM Error: 0x{:08X}", code);
+    report_fw_error_non_fatal(code);
+
+    // Zeroize the crypto blocks.
+    unsafe {
```

Comment:
@mhatrevi Is this actually necessary? I would prefer not to go behind the back of the driver instances in RomEnv to change peripheral state, since they're going to be used later on.

These unsafe methods are really only intended for use in fatal trap handlers, where we can't use the safe interface because we don't have access to it.

I think there's two ways forward:

- Don't zeroize here, since these aren't fatal FIPS errors, and the drivers should already zeroize their contents.

- If zeroization here is a requirement, use a safe API on the drivers in the RomEnv.

Reply:
I agree. I do not think that it is a requirement to zeroize here. We are able to transfer the control to FMC exactly like cold boot without zeroizing. We are still transferring the control here, don't see why we should need to zeroize.

Reply:
I would suggest going with the first option since this is a soft failure.

Reply:
Updated!!